### DISCUSSION TITLE
[Unit Test] MKL Native Provider not found

### DISCUSSION THREAD

**Discussion by gandarez** [mathnet/mathnet-numerics#818] [Q&A] [answered]
I've set NATIVE and MKL on building however I'm getting the message below on "UnitTests" project. How can I write a test using MKL provider?

```
OneTimeSetUp: System.NotSupportedException : MKL Native Provider not found.
  ----> System.DllNotFoundException : Unable to load DLL 'MathNet.Numerics.MKL.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
  Exception doesn't have a stacktrace
```

relates to #453

**Comment by cdrnet** [comment]
You can build and run tests against the MKL provider by running `./build.sh MklWinTest` (or `MklWin64Test` for x64 only, 32 for x86). However, you need to have the Intel Compiler and Intel MKL licensed and installed locally in order to build the MKL provider yourself.

**Comment by cdrnet** [comment, marked as answer]
You could also [download](http://1drv.ms/1NlUeDT) an existing MKL provider release zip archive, extract it to the right directory (out/MKL/Windows/) and then use `UnitTests-MKL.csprj` to build and run the unit tests.
